### ISSUE TITLE
"H * solution = y" not "solution * H = y"

### ISSUE THREAD

**Issue by MohamedNennouche** [Salahberra2022/deep-unfolding#26]
In utils.py line 66: (in gen_linear)

solution = torch.normal(torch.zeros(bs, n), 1.0).to(device).detach()
y = solution @ ht.detach()

This part of the code should be modified as follows:

solution = torch.normal(torch.zeros(m, bs), 1.0).to(device).detach()
y = ht.detach() @ solution 

This corresponds to an Ax=b problem where the objective is to find x. This change will directly induce modifications in the other two files by inverting the matrix multiplications and inverting n and m in the matrix sizes, but the algorithm will remain the same.

What do you think ?

**Comment by nunofachada** [comment]
Hi Mohamed, I hope everything is well.
I'll take a look at this on Thursday, Friday at most.
Thank you!

**Comment by MohamedNennouche** [comment]
thank you very much Nuno, I'll try to add some pull requests for at night. I review the rest of this part and correct the last part in deep_unfolding

**Comment by Salahberra2022** [comment]
Thank you, Nuno and Mohammed, for coming back.
